server/src/test/java/io/crate/expression/predicate/DistinctFromPredicateTest.java

```diff
@@ -0,0 +1,296 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.predicate;
+
+import static io.crate.testing.Asserts.isFunction;
+import static io.crate.testing.Asserts.isLiteral;
+
+import java.util.List;
+
+import org.junit.Test;
+
+import io.crate.expression.operator.DistinctFromPredicate;
+import io.crate.expression.scalar.ScalarTestCase;
+import io.crate.expression.symbol.Literal;
+import io.crate.types.ArrayType;
+import io.crate.types.DataTypes;
+
+/*
+A <distinct predicate> tests two values to see whether they are distinct and
+returns either ``TRUE`` or ``FALSE``.
+ */
+public class DistinctFromPredicateTest extends ScalarTestCase {
```

Comment:
`IS NOT DISTINCT` tests do not seem necessary, `NotPredicate` should be able to handle any kind of child expressions.

Similarly, I am not sure if the array variants are necessary but I can see that you referenced `EqOperatorTest`.

Reply:
I agree that some of these tests seem a bit redundant as the cases are covered by the `DataType.compare` implementation/tests

I'd reduce it as follows:
- Keep the ones specific to the null handling of distinct from
- Merge the `<value> is (not) distinct from <value>` cases into a single case that uses `DataTypeTesting.randomType()` + `DataTypeTesting.getDataGenerator`